library/ssl_tls.c

```diff
@@ -8920,4 +8920,76 @@ int mbedtls_ssl_get_key_exchange_md_tls1_2( mbedtls_ssl_context *ssl,
 #endif /* MBEDTLS_SSL_PROTO_TLS1 || MBEDTLS_SSL_PROTO_TLS1_1 || \
           MBEDTLS_SSL_PROTO_TLS1_2 */
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2_OR_EARLIER) && defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+static void ssl_update_checksum_start( mbedtls_ssl_context* ssl,
+                                       const unsigned char* buf, size_t len )
+{
+    if( ssl->minor_ver==MBEDTLS_SSL_MINOR_VERSION_4
+        && ssl->major_ver == MBEDTLS_SSL_MAJOR_VERSION_3)
```

Comment:
We don't need to check for the major version, it's always the same.
```suggestion
    if( ssl->minor_ver==MBEDTLS_SSL_MINOR_VERSION_4 )
```